### ISSUE TITLE
Regression - No valid credential sources found for AWS Builder.

### ISSUE THREAD

**Issue by JustAMan** [hashicorp/packer#5986]
This is a follow-up for #5901
It was stated there that the problem was fixed in `1.2.1`, though I'm still experiencing it (but a version attached there as a pre-release binary works just fine).

- **Packer version**: 1.2.1 (downloaded as single binary from releases web page).
- **Host platform**: Windows 10 
- **Debug log output**: https://gist.github.com/JustAMan/28413ad29ed8d06c951bb77da8f06656
Note that I've slightly edited the log file (replaced some specific paths and proxy URLs)
- **Configuration file to reproduce**: as for specific example - I basically have a config that uses EBS builder with (so far) hardcoded credentials. Simplest version is here: https://gist.github.com/JustAMan/11bd33d96a1520e146da5b49a9530ac0

As a note for investigators: I also have `terraform.exe` placed next to `packer.exe`, and I have a Terraform config file in the directory I have packer config. I've used Terraform before to create a few nodes on Amazon (using same credentials, also hardcoded in Terraform config for now).

**Comment by SwampDragons** [comment]
cc @mwhooker -- looks like your fix didn't solve for all situations

**Comment by JustAMan** [comment]
Forgot to add - I'm using `packer 1.2.0` for now, and for my case it works (I did not have to revert to `1.1.3` as in previous issue).

**Comment by nathanpalmer** [comment]
Just ran into the same thing here on an alpine linux host. 

> No valid credential sources found for AWS Builder

Downgrading to 1.2.0 and the error went away.

**Comment by koitsu** [comment]
Encountering this on FreeBSD 11.x as well.  Verified IAM policies are correct per https://www.packer.io/docs/builders/amazon.html#iam-task-or-instance-role and so on.

Packer 1.2.0 works fine.

**Comment by mwhooker** [comment]
Sorry about this, folks. Would anyone here mind testing the code in #6019?

**Comment by koitsu** [comment]
Tested #6019 / branch fix-aws-credentials. Works on FreeBSD 11.x with Go 1.10. :+1:

**Comment by JustAMan** [comment]
@mwhooker could you please provide Windows binary? I don't have Go installed currently on my laptop but I'm willing to check.

**Comment by SwampDragons** [comment]
@JustAMan here's a windows binary of the master branch. 
[packer.zip](https://github.com/hashicorp/packer/files/1826087/packer.zip)

**Comment by JustAMan** [comment]
@SwampDragons I can confirm that it fixes my problem, too.

**Comment by SwampDragons** [comment]
Excellent, thanks @JustAMan

**Comment by banza** [comment]
Hello, in macOS the same issue. Packer 1.2.1

**Comment by SwampDragons** [comment]
@banza the fix will be released this Friday; do you want me to build you a macOS binary so that you can verify the release will fix your issue?

**Comment by banza** [comment]
@SwampDragons yes, it would be great! Thank you.

**Comment by SwampDragons** [comment]
[packer.tar.gz](https://github.com/hashicorp/packer/files/1838425/packer.tar.gz)
@banza here you go! This is a build of the master branch; let me know if your problem still exists.

**Comment by banza** [comment]
@SwampDragons sorry for delay, yep, my problem solved, thank you!

**Comment by nathanhleung** [comment]
Hi, I just wanted to chime in and say that I'm facing the same issue on my Windows 10 machine.

I `choco install`'d Packer 1.2.1 on my Windows 10 machine today and tried to follow along with the template given here: https://www.packer.io/intro/getting-started/build-image.html

I kept getting the "No valid credential sources found for AWS Builder" and tried using variations of `-var` and `-var-file` on both PowerShell and CMD but kept getting the same error.

I downgraded to 1.1.3 (`choco install packer --version 1.1.3 --allow-downgrade`) and it worked like a charm.

I'm very new to Packer so I'm not sure where my log files are, but if it would help I'd be happy to provide them - just let me know how I can get to them.
